### PR TITLE
AA-4903: Log Source Update Request 

### PR DESCRIPTION
**Pull request by cclifton-armor**

### CODE REVIEW

**Comment by straineanu** [comment]
@cclifton-armor I ran `pre-commit run -a` because the jenkins build failed on pre-commit. That auto-resolved the issue and I just pushed the commit. 👍

Normally `pre-commit` on its own is fine because it will check the files you have modified.

File: src/logs/log-source-update-request.struct.apib [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
++ orgid (number, required)
+    The unique identifier of organization.
 
-+ version (number, required)
-    The version of this log source (used for concurrency checking). Pass this value with any subsequent delete request.
++ logSourceTypeId (number, required)
+    The unique identifier of the log source type this log source was provisioned as.
+
++ logSourceProtocolId (number, required)
+    The unique identifier of the log source protocol this log source uses.
```
Comment (straineanu) [inline src/logs/log-source-update-request.struct.apib]: @cclifton-armor `logSourceTypeId` and `logSourceProtocolId` should be removed. They can't actually be updated. The API/python code can use the values retrieved from DynamoDB if it needs to include them in the SNS message.

File: src/logs/log-source-update-request.struct.apib [hunk 2]
```diff
@@ -1,10 +1,16 @@
 ## LogSourceUpdateRequest (object)
 
-+ hostname (Hostname, required)
-    The hostname of the enitty for which the log source will process logs. This must be universally unique!
++ logSourceId (Guid, required)
+    The unique identifier of the log source.
 
-+ acls (array[CIDR], required)
-    An array of IPv4 network subnets in CIDR notation that are permitted to send logs to this endpoint.
++ orgid (number, required)
```
[the remark this answers is not in the document]
Comment (cclifton-armor) [reply src/logs/log-source-update-request.struct.apib]: The Changes have been made to remove orgid

File: src/logs/log-source-update-request.struct.apib (hunk 1 above)
Comment (cclifton-armor) [reply src/logs/log-source-update-request.struct.apib]: The Changes have been made to remove logSourceProtocolId and logSourceTypeId

File: src/logs/log-source-update-request.struct.apib [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
+    The unique identifier of the log source.
 
-+ acls (array[CIDR], required)
-    An array of IPv4 network subnets in CIDR notation that are permitted to send logs to this endpoint.
++ options (object)
+    An object map of configuration properties when the log source protocol requires additional configuration options.
 
-+ version (number, required)
-    The version of this log source (used for concurrency checking). Pass this value with any subsequent delete request.
++ version (number)
```
Comment (straineanu) [inline src/logs/log-source-update-request.struct.apib]:
```suggestion
+ version (number, required)
```

File: src/logs/log-source-update-request.struct.apib [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
+    The version of this log Source (used for concurrency checking). Pass this value with any subsequent update request.
```
Comment (straineanu) [inline src/logs/log-source-update-request.struct.apib]:
```suggestion
    The version of this log source (used for concurrency checking). Pass this value with any subsequent update request.
```

File: src/logs/log-source-update-request.struct.apib [hunk 5]
```diff
@@ -1,10 +1,10 @@
 ## LogSourceUpdateRequest (object)
 
-+ hostname (Hostname, required)
-    The hostname of the enitty for which the log source will process logs. This must be universally unique!
++ logSourceId (Guid, required)
+    The unique identifier of the log source.
 
-+ acls (array[CIDR], required)
-    An array of IPv4 network subnets in CIDR notation that are permitted to send logs to this endpoint.
++ options (object)
```
Comment (straineanu) [inline src/logs/log-source-update-request.struct.apib]:
```suggestion
+ options (object, required)
```

File: .nvmrc [hunk 6]
```diff
@@ -0,0 +1 @@
+8.16.0
```
Comment (straineanu) [inline .nvmrc]: This is awesome. Thanks for including and updating the README!🔥

File: src/logs/log-source-update-request.struct.apib (hunk 3 above)
Comment (cclifton-armor) [reply src/logs/log-source-update-request.struct.apib]: The Required flag has been added.

File: src/logs/log-source-update-request.struct.apib (hunk 4 above)
Comment (cclifton-armor) [reply src/logs/log-source-update-request.struct.apib]: Source has been updated to be lowercase.

File: src/logs/log-source-update-request.struct.apib (hunk 5 above)
Comment (cclifton-armor) [reply src/logs/log-source-update-request.struct.apib]: The Required flag has been added.

**Pull request merged**